src/impl/core/dappletEngine.ts

```diff
@@ -2,12 +2,13 @@ import { DappletFrameExecutor } from "./dappletFrameExecutor";
 import { DappletRequest } from "../../types/dappletRequest";
 import { DappletProvider } from "../../interfaces/dappletProvider";
 import { FrameStatus } from "../../types/statusEnum";
+import { FeaturesRegistry } from '../featuresRegistry';
 
 export class DappletEngine {
     private _frameExecutors: DappletFrameExecutor[] = [];
     private _statusChangedHandler: (readiness: number) => void = () => {};
 
-    constructor(dappletRequest: DappletRequest) {
+    constructor(dappletRequest: DappletRequest, private featuresRegistry: FeaturesRegistry) {
```

Comment:
I would probably use a dependency to the whole `DappletContext` instead of `FeatureRegistry` only.
The will be more services from the Context used by Engine. For example to access external resources.